File: carshop-android/app/src/main/java/com/carshop/android/data/dto/Product.kt
package com.carshop.android.data.dto

import com.squareup.moshi.Json

data class Product(
    val id: Int,
    @Json(name = "category_id") val categoryId: Int,
    @Json(name = "category_name") val categoryName: String?,
    val title: String,
    // SPEC §5:"physical" 或 "service_voucher"
    @Json(name = "product_type") val productType: String,
    val price: Int,
    @Json(name = "original_price") val originalPrice: Int?,
    val spec: String?,
    @Json(name = "main_image_url") val mainImageUrl: String,
    val description: String,
    @Json(name = "on_sale") val onSale: Boolean,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "updated_at") val updatedAt: String,
)

File: carshop-admin/src/api/product.ts
import { apiClient, unwrap, type Page } from './client';

export type ProductType = 'physical' | 'service_voucher';

export interface Product {
  id: number;
  category_id: number;
  category_name: string;
  title: string;
  product_type: ProductType;
  price: number; // 分
  original_price: number | null;
  spec: string | null;
  main_image_url: string;
  description: string;
  on_sale: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProductInput {
  category_id: number;
  title: string;
  product_type: ProductType;
  price: number;
  original_price?: number | null;
  spec?: string | null;
  main_image_url: string;
  description?: string;
  on_sale: boolean;
}

export interface ProductListParams {
  category_id?: number;
  on_sale?: boolean;
  page?: number;
  page_size?: number;
}

export const productApi = {
  list: (params: ProductListParams = {}) =>
    unwrap<Page<Product>>(apiClient.get('/admin/products', { params })),
  detail: (id: number) => unwrap<Product>(apiClient.get(`/admin/products/${id}`)),
  create: (body: ProductInput) =>
    unwrap<Product>(apiClient.post('/admin/products', body)),
  update: (id: number, body: Partial<Omit<ProductInput, 'on_sale'>>) =>
    unwrap<Product>(apiClient.put(`/admin/products/${id}`, body)),
  setOnSale: (id: number, on_sale: boolean) =>
    unwrap<Product>(apiClient.patch(`/admin/products/${id}/on_sale`, { on_sale })),
  remove: (id: number) =>
    unwrap<{ id: number; deleted: boolean }>(apiClient.delete(`/admin/products/${id}`)),
};
